mobile/app/src/main/java/com/yybb/myapplication/presentation/ui/utils/GraphView.kt

```diff
@@ -0,0 +1,402 @@
+        color = Color.BLACK
+        strokeWidth = 5f
+        isAntiAlias = true
+    }
+
+    private val edgeTextPaint = Paint().apply {
+        color = Color.MAGENTA
+        textSize = 41f
+        isAntiAlias = true
+    }
+
+    // Pan & Zoom
+    var scaleFactor = 1f
+    private var minScaleFactor = 0.5f
+    private val scaleDetector = ScaleGestureDetector(context, ScaleListener())
+    var offsetX = 0f
+    var offsetY = 0f
+    private var isDragging = false
+
+    init {
+        val rnd = kotlin.random.Random.Default
+        val canvasWidth = 4000f
+        val canvasHeight = 4000f
+        val radius = 120f
+        val clusterRadius = 400f
+
+        // Step 1: Create nodes
+        for (i in 1..5) {
+            nodes.add(
+                Node(
+                    id = i,
+                    x = rnd.nextFloat() * canvasWidth,
+                    y = rnd.nextFloat() * canvasHeight,
+                    radius = radius,
+                    title = "Node $i",
+                    description = "This is node $i"
+                )
+            )
+        }
+
+        // Step 2: Generate edges (connections)
+        val nodeCount = nodes.size
+        val edgeSet = mutableSetOf<Pair<Int, Int>>()
+        while (edgeSet.size < 3) {
+            val fromIdx = rnd.nextInt(nodeCount)
+            var toIdx = rnd.nextInt(nodeCount)
+            if (fromIdx == toIdx) continue
+            val pair = if (fromIdx < toIdx) Pair(fromIdx, toIdx) else Pair(toIdx, fromIdx)
+            if (edgeSet.add(pair)) {
+                edges.add(
+                    Edge(
+                        from = nodes[fromIdx],
+                        to = nodes[toIdx],
+                        description = "Edge from ${nodes[fromIdx].title} to ${nodes[toIdx].title}"
+                    )
+                )
+            }
+        }
+
+        // Step 3: Reposition connected nodes closer together
+        for (edge in edges) {
+            val from = edge.from
+            val to = edge.to
+
+            val angle = rnd.nextFloat() * 2 * Math.PI
+            val distance = clusterRadius * (0.5f + rnd.nextFloat() * 0.5f)
+            to.x = from.x + (distance * kotlin.math.cos(angle)).toFloat()
+            to.y = from.y + (distance * kotlin.math.sin(angle)).toFloat()
+        }
+
+        relaxLayout()
+    }
+
+    override fun onDraw(canvas: Canvas) {
+        super.onDraw(canvas)
+
+        canvas.save()
+        canvas.translate(offsetX, offsetY)
+        canvas.scale(scaleFactor, scaleFactor)
+
+        // Draw edges
+        for (edge in edges) {
+            canvas.drawLine(edge.from.x, edge.from.y, edge.to.x, edge.to.y, edgePaint)
+            val midX = (edge.from.x + edge.to.x) / 2
+            val midY = (edge.from.y + edge.to.y) / 2
+            canvas.drawText(edge.description, midX, midY, edgeTextPaint)
+        }
+
+        // Paints
+        val borderPaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
```

Comment:
We should move this variables to top-level as private, right now every time onDraw function is called we declare these 3 variables again which may cause laggy UI as graph gets bigger.